edit/edit_test.go

```diff
@@ -673,3 +676,51 @@ x = 2`,
 		}
 	}
 }
+
+type testCase struct {
+	inputRoot, inputTarget                       string
+	expectedBuildFile, expectedPkg, expectedRule string
+}
+
+func runTestInterpretLabelForWorkspaceLocation(t *testing.T, buildFileName string) {
+	tmp, err := ioutil.TempDir("", "")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(tmp)
+	if err := os.MkdirAll(filepath.Join(tmp, "a", "b"), 0755); err != nil {
+		t.Fatal(err)
+	}
+	// todo(aljoscha): See if we should hardcode "WORKSPACE" here or should use wspace.workspaceFile
```

Comment:
You have two pending TODOs, are you going to keep them? If so, it's better to file corresponding bugs and use links for them instead of your username, this way it'll be much easier to track them.

Reply:
Ah yes, sorry about that. I meant them mostly as questions for the reviewer or someone familiar with the code

So for the first one: should I leave `"WORKSPACE"` here and assume that this will always be the filename of workspace files?

For the second one: I don't know when the second part of the `InterpretLabelForWorkspaceLocation` function would be used. If I remove the second part of the method, that is those lines:
```
if isFile(pkg) {
	// allow operation on other files like WORKSPACE
	buildFile = pkg
	pkg = filepath.Join(relativePath, filepath.Dir(pkg))
	return
}
if pkg != "" {
	buildFile = pkg + "/BUILD"
	if !isFile(buildFile) {
		// try it with the .bazel extension
		buildFile += ".bazel"
	}

} else {
	buildFile = "BUILD"
	if !isFile(buildFile) {
		// try it with the .bazel extension
		buildFile += ".bazel"
	}
}
pkg = filepath.Join(relativePath, pkg)
```
`bazel test /...` still passes. Which doesn't mean that we don't need them, it might just be missing test coverage.

Reply:
1. I think it's ok to hardcode the "WORKSPACE" string for now.

2. The second part of the function will be executed if the following check doesn't pass: `strings.HasPrefix(target, "//")`, i.e. the label is not absolute but relative, e.g. `":a"` or `":a/b"`.

Reply:
Thanks! And I'm sorry, this is taking up quite a bit of your time.

I understand the flow of the function, I wasn't sure about the behaviour because it was not previously covered by tests. And the existing behaviour is a bit weird (or doesn't throw errors). For example:
 - `/path/to/WORKSPACE:foo` gives (`"/path/to/WORKSPACE"`, `"/path/to"`, `"foo"`)
 - `/path/to/WORKSPACE` gives (`"BUILD"`, `""`, `"/path/to/WORKSPACE"`)
(the tuple is the result of `InterpretLabelForWorkspaceLocation`, which is `(buildFile, pkg, rule)`

The first one seems (is?) correct, the second result seems to be bogus, but this is what the existing code spits out.

Reply:
I think it worth investigating, but right now it's safer to leave it as it is.